ack_vuln.py

```diff
@@ -0,0 +1,249 @@
+
+    parser.add_argument("--encoding", dest="csv_encoding",
+                        help="CSV file encoding, default is %s" % cvconfig.csv_encoding)
+
+    parser.add_argument("--proxy", dest="proxy", default=cvconfig.proxy,
+                        help="Specified the proxy to use (default: %s)" % cvconfig.proxy)
+
+    args = parser.parse_args()
+
+    # Handle Cyber Vision configuration
+    token = set_conf(args.token, cvconfig.token)
+    center_ip = set_conf(args.center_ip, cvconfig.center_ip)
+    center_port = set_conf(args.center_port, cvconfig.center_port)
+    csv_encoding = set_conf(args.csv_encoding, cvconfig.csv_encoding)
+    csv_delimiter = set_conf(args.csv_delimiter, cvconfig.csv_delimiter)
+    proxy = set_conf(args.proxy, cvconfig.proxy)
+
+    # Handle default values
+    ack_comments = set_conf(args.ack_comments, cvconfig.ack_comments)
+
+    if not token or not center_ip:
+        print("TOKEN and CENTER_IP are mandatory, check cvconfig.py or us --token/--center-ip")
+
+    if args.command_export:
+        return export_vulnerable_devices(center_ip, center_port, token, proxy, args.filename, csv_delimiter, csv_encoding, args.cve_id, args.preset_name)
+    elif args.command_ack_by_preset:
+        return ack_vulnerabilities_by_preset(center_ip, center_port, token, proxy, args.cve_id, args.preset_name, ack_comments)
+    elif args.command_ack_by_devices:
+        return ack_vulnerabilities_by_devices(center_ip, center_port, token, proxy, args.filename, csv_delimiter, args.cve_id, ack_comments)
+    
+    parser.print_help()
+
+
+def set_conf(arg, conf):
+    if arg and arg != conf:
+        return arg
+    return conf
+
+
+def build_device_row(row, d):
+    try:
+        row['device-id'] = d['id']
+        row['device-name'] = d['originalLabel']
+        row['device-ip'] = d['ip']
+        row['device-mac'] = d['mac']
+        row['device-custom-name'] = d['customLabel']
+        row['device-riskscore'] = d['riskScore']
+    except KeyError as e:
+        print("KeyError: {} for device {}".format(e, row['device-id']))
+    except Exception as e:
+        print(e)
+
+
+def write_devices(filename, csv_encoding, csv_delimiter, devices):
+    with open(filename, 'w', encoding=csv_encoding) as csvfile:
+        fieldnames = ['device-id', 'device-mac', 'device-ip', 'device-name', 'device-custom-name', 'device-tags', 'device-riskscore']
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames, delimiter=csv_delimiter)
+        writer.writeheader()
+        for d in devices:
+            row = {}
+            build_device_row(row, d)
+            writer.writerow(row)
+        print(f"LOG: Exported {len(devices)} into '{filename}'")
+
+
+def export_vulnerable_devices(center_ip, center_port, token, proxy, filename, csv_delimiter, csv_encoding, cve_id, preset_name):
+    if not preset_name:
+        print(f"WARN: --preset-name not passed, By default \"All data\" preset is considered")
```

Comment:
Again, here all data is used directly in the string - should we use the default value from cvconfig.py instead ?